diff --git a/plugins/views/views_php_plugin_pager.inc b/plugins/views/views_php_plugin_pager.inc
index 3a418e9..115920a
--- a/plugins/views/views_php_plugin_pager.inc
+++ b/plugins/views/views_php_plugin_pager.inc
@@ -36,14 +36,17 @@ class views_php_plugin_pager extends views_php_plugin_wrapper  {
         }
       }
     }
-
-    $this->wrapped->total_items = count($this->wrapped->view->result);
+    if (!isset($this->wrapped->total_items)) {
+      $this->wrapped->total_items = count($this->wrapped->view->result);
+    }
     $this->wrapped->update_page_info();

     $item_per_page = $this->wrapped->get_items_per_page();
     if ($item_per_page > 0) {
       $offset = $this->wrapped->get_current_page() * $item_per_page + $this->wrapped->get_offset();
-      $this->wrapped->view->result = array_slice($this->wrapped->view->result, $offset, $item_per_page);
+      if (isset($this->wrapped->view->result)) {
+        $this->wrapped->view->result = array_slice($this->wrapped->view->result, $offset, $item_per_page);
+      }
     }
     $this->wrapped->post_execute($result);
   }
